### DISCUSSION TITLE
Are we able to scope a Relation Manager query?

### DISCUSSION THREAD

**Discussion by ExpertMediaDesign** [filamentphp/filament#6202] [Help]
I have an unusual setup: I have multiple Resources from a single Model (i.e. Document Model with DocumentsResource, DocGridResource, DocSheetResource, etc.). In those "subresources", we just scope getEloquentQuery() & they work great:
```
public static function getEloquentQuery(): Builder
{
     return parent::getEloquentQuery()->ofType('1');
}
```

I'd like to be able to scope the Relation Manager for the "subresources" in a similar way, though it seems getEloquentQuery() isn't available. Is there another approach I should take here? 

Alternatively, is there a recommended way to insert a custom page/table to the Relation Manager area on a resource?

**Comment by danharrin** [comment]
On relation managers there is `getTableQuery()`

**Reply by sakibulalam** [reply]
isn't this deprecated in v3? should this be the latest approach?

```php

class PostsRelationManager extends RelationManager
{
    public function table(Table $table): Table
    {
        return PostResource::table($table)
            ->query(fn() => $this->ownerRecord->isPublished());
    }
}
```

**Reply by leandrocfe** [reply]
https://filamentphp.com/docs/3.x/panels/resources/relation-managers/#customizing-the-relation-manager-eloquent-query
